### PR TITLE
feat: query the last existing snapshot

### PR DESCRIPTION
**Pull request by politeWall**

### CODE REVIEW

File: a.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+extract_txhash() { awk -F 'txhash: ' '/txhash:/{print $2; exit}'; }
```
Comment (cosmic-vagabond) [inline a.sh:1]: @politeWall please move this file to a more comprehensive folder and file name

File: a.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          cosmwasm/workspace-optimizer:0.14.0
+
+# store and init trade shield contract
+# txhash=$(elysd tx wasm store artifacts/trade_shield_contract.wasm $OPTIONS | extract_txhash)
+# sleep 10
+# codeid=$(elysd q tx $txhash | extract_code_id)
+# txhash=$(elysd tx wasm init $codeid '{}' $OPTIONS  --label "Contract" --admin validator | extract_txhash)
+# sleep 10
+# addr=$(elysd q tx $txhash | extract_contract_address)
+addr=elys1qeyjez6a9dwlghf9d6cy44fxmsajztw257586akk6xn6k88x0gus3ugsrl
```
Comment (cosmic-vagabond) [inline a.sh:19]: @politeWall re-enable the commited code as we want to be able to re-use the entire script

File: contracts/account-history-contract/src/entry_point/query.rs [hunk 3]
```diff
@@ -14,5 +15,19 @@ pub fn query(deps: Deps<ElysQuery>, env: Env, msg: QueryMsg) -> StdResult<Binary
...
+                .collect::<Vec<(String, Vec<AccountSnapshot>)>>();
+            list
+        }),
+        LastSnapshot { user_address } => to_json_binary(&{
+            let snapshots = HISTORY.load(deps.storage, &user_address)?;
+            match snapshots.last().cloned() {
+                Some(expr) => expr,
+                None => return Err(StdError::not_found("account snapshot")),
+            }
+        }),
```
Comment (cosmic-vagabond) [inline contracts/account-history-contract/src/entry_point/query.rs:31]: @politeWall we actually need to have another query that returns the list of liquid assets and total balance of liquid assets alone based on the latest snapshot

File: contracts/account-history-contract/src/tests.rs [hunk 4]
```diff
@@ -22,19 +22,22 @@ fn history() {
...
+            2,
+        ),
+        OracleAssetInfo::new(
+            "uelys".to_string(),
+            "UELYS".to_string(),
+            "".to_string(),
+            "".to_string(),
+            2,
+        ),
+    ];
```
Comment (cosmic-vagabond) [inline contracts/account-history-contract/src/tests.rs:40]: @politeWall any reason why you still need this as you are using asset profile to retrieve decimals value?

**Review by cosmic-vagabond — CHANGES_REQUESTED** [review]

File: contracts/account-history-contract/src/tests.rs (hunk 4 above)
Comment (politeWall) [reply contracts/account-history-contract/src/tests.rs:40]: The reason why is to have a simpler way to mock the "getEntry" query.

**Review by cosmic-vagabond — APPROVED** [review]

**Pull request merged**